components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java

```diff
@@ -58,11 +58,14 @@ public static TenantRegistryLoader getTenantRegistryLoader() {
         return tenantRegistryLoader;
     }
 
+    public static void setTenantRegistryLoader(TenantRegistryLoader tenantRegistryLoader) {
```

Comment:
From where this method is called?
It is not safe to set a static variable via public method as anyone can set to any value , thus system can break or can lead to race conditions.

Reply:
This seems to be an existing method. After the refactoring, it is showing as a new method. At the moment it is called from the service component when it gets a refference to `TenantRegistryLoader` service instance.

Reply:
ack